lib/widget/toggle_button.dart

```diff
@@ -0,0 +1,303 @@
+            entries.add(
+              SelectOneItem(
+                value: element['value'],
+                label: element['label']?.toString(),
+                icon: Utils.getIcon(element['icon']),
+                isIcon: isIconExist(values),
+              ),
+            );
+          }
+        }
+        // simply use the value
+        else {
+          entries.add(SelectOneItem(value: element));
+        }
+      }
+    }
+    _controller.items = entries;
+
+    // ensure that the value is still correct
+    if (!isValueInItems()) {
+      _controller.maybeValue = null;
+    }
+  }
+
+  bool isIconExist(List v) {
+    for (var e in v) {
+      if (e['icon'] != null) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  void calculateIconSize(List v) {
+    List iconSize = [];
+    _controller.gap = 0;
+    for (var e in v) {
+      if (e is Map) {
+        if (e['icon'] != null) {
+          if (e['icon']['size'] != null) {
+            iconSize.add(e['icon']['size']);
+          }
+        }
+      }
+    }
+    if (iconSize.isNotEmpty) {
+      _controller.gap =
+          iconSize.reduce((curr, next) => curr > next ? curr : next);
+    } else {
+      _controller.gap = 0;
+    }
+  }
+
+  void onSelectionChanged(dynamic value) {
+    setProperty('value', value);
+  }
+}
+
+class EnsembleToggleButtonState extends WidgetState<EnsembleToggleButton> {
+  List<ToggleItem>? _items = [];
+
+  @override
+  void initState() {
+    super.initState();
+    _setValue();
+  }
+
+  void _setValue() {
+    if (widget.controller.items != null) {
+      final value = widget.getValue();
+      final valueIndex = widget.controller.items
+          ?.indexWhere((element) => element.value == value);
+      widget.controller.items!.asMap().forEach((index, value) {
+        _items!.add(
+          ToggleItem(
+            label: value.label,
+            value: value.value,
+            icon: value.icon,
+            isIcon: value.isIcon,
+            isSelected: (valueIndex != null && valueIndex != -1)
+                ? index == valueIndex
+                : false,
+          ),
+        );
+      });
+    }
+  }
+
+  @override
+  Widget buildWidget(BuildContext context) {
+    if (_items == null || _items!.isEmpty) return const SizedBox.shrink();
+
+    final controller = widget.controller;
+
+    Widget? rtn = ToggleButtons(
+      borderRadius: const BorderRadius.all(Radius.circular(8)),
+      color: controller.color,
+      selectedColor: controller.selectedColor,
+      fillColor: controller.selectedBackgroundColor,
+      borderColor: controller.borderColor,
+      selectedBorderColor: controller.selectedBorderColor,
+      constraints: const BoxConstraints(
+        minHeight: 40.0,
+        minWidth: 80.0,
+      ),
+      children: List.generate(
+        _items!.length,
+        (index) => _buildWidget(_items![index]),
+      ),
+      isSelected: [
+        ..._items!.map((e) => e.isSelected).toList(),
+      ],
```

Comment:
nit: extra list being created here?

Reply:
Yeah. For the isSelected, we need to pass an array of booleans for the selected/unselected state of an item